docker-compose.yml

```diff
@@ -1,35 +1,120 @@
-version: "3"
+version: "3.4"
+
+x-healthcheck: &healthcheck-interval
+  interval: 1s
+  timeout: 1s
+  start_period: 10s
+  # TODO: https://github.com/docker/compose/issues/10461
+  # interval: 10s
+  # start_interval: 1s
+
+x-environment: &environment
+    RUST_BACKTRACE: true
+
+    DOCSRS_PREFIX: /opt/docsrs/prefix
+
+    DOCSRS_DATABASE_URL: postgresql://cratesfyi:password@db
+    DOCSRS_MIN_POOL_SIZE: 2
+    DOCSRS_MAX_POOL_SIZE: 10
+
+    DOCSRS_STORAGE_BACKEND: s3
+
+    S3_ENDPOINT: http://s3:9000
+    AWS_ACCESS_KEY_ID: cratesfyi
+    AWS_SECRET_ACCESS_KEY: secret_key
+
+    DOCSRS_RENDER_THREADS: 2
+
+    DOCSRS_RUSTWIDE_WORKSPACE: /opt/docsrs/rustwide
+    DOCSRS_DOCKER: true
+    DOCSRS_DOCKER_IMAGE: ghcr.io/rust-lang/crates-build-env/linux-micro
+    DOCSRS_BUILD_CPU_LIMIT: 2
+    DOCSRS_INCLUDE_DEFAULT_TARGETS: false
+
+x-builder: &builder
+    build:
+        context: .
+        dockerfile: ./dockerfiles/Dockerfile
+        target: build-server
+    depends_on:
+        - db
+        - s3
+    environment: *environment
+    env_file:
+        - .docker.env
```

Comment:
why can't we use `.env` here?

Reply:
As mentioned below, if `.env` is being used for local settings those are not all appropriate for setting inside docker. It's also loaded by `sqlx` internally through `dotenvy`, so it can't be used to hold settings for docker only and leave local settings to `.envrc` or something (as I do it with `direnv`).